### PR TITLE
pipeline-gcp: add yml and enable cd

### PR DESCRIPTION
**Pull request by Enigo**
# Link to GitHub repository

https://github.com/jenkinsci/pipeline-gcp-plugin

# When modifying release permission

List the GitHub usernames of the users who should have commit permissions below:
- @Enigo 

This is needed in order to cut releases of the plugin or component.

If you are modifying the release permission of your plugin or component, fill out the following checklist:

```[tasklist]
### Release permission checklist (for submitters)
- [x] The usernames of the users added to the "developers" section in the .yml file are the same the users use to log in to [accounts.jenkins.io](https://accounts.jenkins.io/).
- [x] All users added have logged in to [Artifactory](https://repo.jenkins-ci.org/) and [Jira](https://issues.jenkins.io/) once.
- [ ] I have mentioned an [existing team member](https://github.com/orgs/jenkinsci/teams) of the plugin or component team to approve this request.
```

## When enabling automated releases (cd: true)

Follow the [documentation](https://www.jenkins.io/doc/developer/publishing/releasing-cd/) to ensure, your pull request is set up properly. Don't merge it yet.  
In case of changes requested by the hosting team, an open PR facilitates future reviews, without derailing work across multiple PRs.

### Link to the PR enabling CD in your plugin

https://github.com/jenkinsci/pipeline-gcp-plugin/pull/3

```[tasklist]
### CD checklist (for submitters)
- [x] I have provided a link to the pull request in my plugin, which enables CD according to the documentation. 
```

```[tasklist]
### Reviewer checklist
- [ ] Check that the `$pluginId Developers` team has `Admin` permissions while granting the access.
- [ ] In the case of plugin adoption, ensure that the Jenkins Jira default assignee is either removed or changed to the new maintainer.
- [ ] If security contacts are changed (this includes add/remove), ping the security officer (currently `@Wadeck`) in this pull request. If an email contact is changed, wait for approval from the security officer.
```
There are [IRC Bot commands](https://jenkins.io/projects/infrastructure/ircbot/#issue-tracker-management) for it.

### CODE REVIEW

File: permissions/plugin-pipeline-gcp.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+name: "pipeline-gcp"
+github: &GH "jenkinsci/pipeline-gcp-plugin"
+issues:
+  - github: *GH
+paths:
+  - "org/jenkins-ci/plugins/pipeline-gcp"
```
Comment (NotMyFault) [inline permissions/plugin-pipeline-gcp.yml:7]:
```suggestion
  - "io/jenkins/plugins/pipeline-gcp"
```

File: permissions/plugin-pipeline-gcp.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+developers:
+  - "enigo"
+cd:
+  enabled: true
```
Comment (NotMyFault) [inline permissions/plugin-pipeline-gcp.yml:11]:
```suggestion
```
You can enable that once your plugin is prepared for CD.

**Review by NotMyFault — APPROVED** [review]
Normally this is done automatically, but the process broke.

File: permissions/plugin-pipeline-gcp.yml (hunk 2 above)
Comment (Enigo) [reply permissions/plugin-pipeline-gcp.yml]: but that's what this [PR](https://github.com/jenkinsci/pipeline-gcp-plugin/pull/3) is about, I thought

Comment (NotMyFault) [reply permissions/plugin-pipeline-gcp.yml]: Nope, this just adds upload permissions to artifactory

Comment (Enigo) [reply permissions/plugin-pipeline-gcp.yml]: but I added [workflow/cd.yaml](https://github.com/jenkinsci/pipeline-gcp-plugin/pull/3/files#diff-7820634de9167dbbf6a1ab9e9c02aab3b8b0ca512494aca230074fd13f630ade) there as per the instructions :thinking:

Comment (NotMyFault) [reply permissions/plugin-pipeline-gcp.yml]: There's more to do and PRs are reviewed separately

Comment (Enigo) [reply permissions/plugin-pipeline-gcp.yml]: yeah, I added\removed other things as well
ok, will open a separate PR then, no probs

**Pull request merged**